connectors/preflight_check.py

```diff
@@ -49,6 +54,35 @@ async def run(self):
             if self.es_client is not None:
                 await self.es_client.close()
 
+    async def _check_local_extraction_setup(self):
+        if not self.extraction_config["enabled"]:
```

Comment:
```suggestion
        if not self.extraction_config.get("enabled", False):
```

nit: We could be a bit more defensive here